app/models/invenio_rdm_record_converter.rb

```diff
@@ -89,13 +91,23 @@ def owner_info(generic_file)
     end
   end
 
-  def record_for_export(generic_file)
+  def file_permissions
+    permission_data = Hash.new { |h,k| h[k] = [] }
```

Comment:
Edit: `Hash.new` seems right for here.

This just returns an empty hash in IRB/Rails console :

```
2.3.3 :001 > Hash.new{ |h, k| h[k] = [] }
 => {}
```

I say we replace it with either `Hash.new` or if you want to initialize a hash with an empty array in it you could do something like `hash = {key: []}`

Reply:
It might be my ruby version, but when I try just using `Hash.new`, I was not able construct the value to be an array on the fly. The `h[k] = []` part ensures that the value is an array, so I can use the shovel operator if the data didn't exist yet. Though, yes I could do `hash = {"read": [], "edit": []}`, I want to cover any types of permissions I might not know about.
``` ruby
[1] pry(main)> data = Hash.new
=> {}
[2] pry(main)> data["fizz"] << "buzz"
NoMethodError: undefined method `<<' for nil:NilClass
from (pry):2:in `<main>'
[3] pry(main)> data = Hash.new { |h,k| h[k] = [] }
=> {}
[4] pry(main)> data["fizz"] << "buzz"
=> ["buzz"]
```

Reply:
That's interesting. I wonder what's going on under the hood here? I checked myself and it's working as you described.